src/Compilers/CSharp/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.cs

```diff
@@ -2277,6 +2273,48 @@ public void Bug1095618()
             });
         }
 
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void ConsistentErrorMessageWhenProvidingNullKeyFile()
+        {
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: null);
+            var compilation = CreateCompilationWithMscorlib(string.Empty, options: options).VerifyDiagnostics();
+
+            VerifySignedBitSetAfterEmit(compilation, expectedToBeSigned: false);
+        }
+
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void ConsistentErrorMessageWhenProvidingEmptyKeyFile()
+        {
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: string.Empty);
+            var compilation = CreateCompilationWithMscorlib(string.Empty, options: options).VerifyDiagnostics();
+
+            VerifySignedBitSetAfterEmit(compilation, expectedToBeSigned: false);
+        }
+
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void ConsistentErrorMessageWhenProvidingNullKeyFile_PublicSign()
+        {
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: null, publicSign: true);
+            CreateCompilationWithMscorlib(string.Empty, options: options).VerifyDiagnostics(
+                // error CS8102: Public signing was specified and requires a public key, but no public key was specified.
+                Diagnostic(ErrorCode.ERR_PublicSignButNoKey).WithLocation(1, 1));
+        }
+
+        [Fact]
+        [WorkItem(11497, "https://github.com/dotnet/roslyn/issues/11497")]
+        public void ConsistentErrorMessageWhenProvidingEmptyKeyFile_PublicSigne()
+        {
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, cryptoKeyFile: string.Empty, publicSign: true);
+            CreateCompilationWithMscorlib(string.Empty, options: options).VerifyDiagnostics(
+                // error CS8106: Option 'CryptoKeyFile' must be an absolute path.
```

Comment:
>// error CS8106: Option 'CryptoKeyFile' must be an absolute path. [](start = 16, length = 65)

I believe that we should not be reporting this error for an empty string as we don't report it for null and the other error clearly states that the key was not specified. If it was not specified, it shouldn't be validated.